lib/esp/extensions/active_resource/formats/json_api_format.rb

```diff
@@ -43,7 +43,7 @@ def decode(json)
         Formats.remove_root(parse_json_api(ActiveSupport::JSON.decode(json)))
       end
 
-      private
+      private_class_method
```

Comment:
Is this doing what you expect it to? I believe you have to pass the method names to `private_class_method`.